docs/docs/92-development.md

```diff
@@ -0,0 +1,205 @@
+
+```bash
+cd docs/
+
+yarn install
+
+# build plugins used by the docs
+yarn build:woodpecker-plugins
+
+# start docs with hot-reloading, so you can change the docs and directly see the changes in the browser without reloading it manually
+yarn start
+
+# or build the docs to deploy it to some static page hosting
+yarn build
+```
+
+## Testing & linting code
+
+To test or lint parts of Woodpecker you can run one of the following commands:
+
+```bash
+# test server code
+make test-server
+
+# test agent code
+make test-agent
+
+# test cli code
+make test-cli
+
+# test datastore / database related code like migrations of the server
+make test-server-datastore
+
+# lint go code
+make lint
+
+# lint UI code
+make lint-frontend
+
+# test UI code
+make test-frontend
+```
+
+If you want to test a specific go file you can also use:
+
+```bash
+go test -race -timeout 30s github.com/woodpecker-ci/woodpecker/<path-to-the-package-or-file-to-test>
+```
+
+Or you can open the file inside [VS-Code](#developing-with-vs-code) and run or debug the test by clicking on the inline commands:
+
+![Run test via VS-Code](92-development/vscode-run-test.png)
+
+## Package architecture of Woodpecker
+
+![Woodpecker architecture](92-development/woodpecker-architecture.png)
+
+## Run applications from terminal
+
+If you want to run a Woodpecker applications from your terminal you can use execute following commands from the base of the Woodpecker project. They will execute Woodpecker in a similar way as described in [debugging Woodpecker](#debugging-woodpecker) without the ability to really debug it in your editor.
+
+```bash
+# start server
+$ go run ./cmd/server
+
+# start agent
+$ go run ./cmd/agent
+
+# execute cli command
+$ go run ./cmd/cli [command]
+```
+
+## Add new migration
+
+Woodpecker uses migrations to change the database schema if a database model has been changed. If a developer for example adds a new property `IsKingKong` to the database model of a User in `server/model/` he would need to add a new migration like the following example to a file like `server/store/datastore/migration/123_add_is_king_kong_to_users.go`:
+
+```go
+// Copyright 2021 Woodpecker Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package migration
+
+import (
+	"xorm.io/xorm"
+)
+
+var addIsKingKongToUsers = task{
+	name: "add-king-kong-to-users",
+	fn: func(sess *xorm.Session) error {
+    // TODO
+		return sess.Commit()
+	},
+}
```

Comment:
@6543 Could you add a simple example to this migration. I don't know if the current example makes sense at all as we run xorm.sync at the end anyways, right? 🤔

Reply:
yes adding new col to a table do only need to update the struct in model - if you like to drop or change something then you will need it

Reply:
NO sess.Commit/Begin/... -> it's handled by the caller of migration func!